### PR TITLE
accounts-sso updates

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Update to proper versions. There actually shouldn't be any real changes IIRC.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
Weird, it seems `gsignondPlugins.mail` msg didn't trigger a build.

@GrahamcOfBorg build gsignondPlugins.mail

**Comment by worldofpeace** [comment]
Noticed some minor discrepancies so I went ahead and corrected them.

File: pkgs/development/libraries/gsignond/plugins/lastfm.nix [hunk 1]
```diff
@@ -29,7 +29,7 @@ stdenv.mkDerivation rec {
   PKG_CONFIG_GSIGNOND_GPLUGINSDIR = "${placeholder "out"}/lib/gsignond/gplugins";
 
   meta = with stdenv.lib; {
-    description = "Plugin for the Accounts-SSO gSignOn daemon handles the Last.FM credentials.";
+    description = "Plugin for the Accounts-SSO gSignOn daemon handles the Last.FM credentials";
```
Comment (c0bw3b) [inline pkgs/development/libraries/gsignond/plugins/lastfm.nix]: `Last.FM Plugin for the gSignOn daemon`
or
`This plugin for the Accounts-SSO gSignOn daemon handles Last.FM credentials`

File: pkgs/development/libraries/libsignon-glib/default.nix [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
-  version = "3639a2e90447e4640a03a44972560afe8f61aa48";
+  version = "2.1";
 
-  name = "${pname}-2018-10-24";
+  name = "${pname}-${version}";
 
   outputs = [ "out" "dev" "devdoc" "py" ];
 
   src = fetchgit {
     url = "https://gitlab.com/accounts-sso/${pname}";
```
Comment (c0bw3b) [inline pkgs/development/libraries/libsignon-glib/default.nix]: You could use `fetchFromGitLab` here

Comment (worldofpeace) [reply pkgs/development/libraries/libsignon-glib/default.nix]: They use a submodule for the Signon DBus specification that's shared across their projects. 
`fetchFromGitLab` won't get the submodule.

Comment (c0bw3b) [reply pkgs/development/libraries/libsignon-glib/default.nix]: Ah yes, I forgot it does not understand `fetchSubmodules = true` like `fetchFromGitHub`

Comment (worldofpeace) [reply pkgs/development/libraries/libsignon-glib/default.nix]: Actually just realized the `rev` should be ` rev = "refs/tags/${version}";`

**Comment by worldofpeace** [comment]
Ahh staging-next was merged, so conflicts now because of the derivation changes.

**Pull request merged**